backend/getData.js

```diff
@@ -52,9 +52,51 @@ function getData(internalInstance: Object): DataType {
     children = childrenList(internalInstance._renderedChildren);
   } else if (internalInstance._currentElement && internalInstance._currentElement.props) {
     // This is a native node without rendered children -- meaning the children
-    // prop is just a string or (in the case of the <option>) a list of
-    // strings & numbers.
-    children = internalInstance._currentElement.props.children;
+    // prop is the unfiltered list of children.
+    // This may include 'null' or even other invalid values, so we need to
+    // filter it the same way that ReactDOM does.
+    // We do a simplified filtering for valid children, instead of using
+    // `React.Children` so that we don't pull in the whole React library.
+    const REACT_ELEMENT_TYPE =
```

Comment:
The iteration logic is copied loosely from `traverseAllChildrenImpl` in `ReactChildren.js`:
https://github.com/facebook/react/blob/240b84ed8e1db715d759afaae85033718a0b24e1/src/isomorphic/children/ReactChildren.js#L112-L158